src/Nethermind/Nethermind.Core/Extensions/Int64Extensions.cs

```diff
@@ -3,120 +3,32 @@
 
 using System;
 using System.Buffers.Binary;
+using System.Numerics;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 using Nethermind.Int256;
 
 namespace Nethermind.Core.Extensions;
 
 public static class Int64Extensions
 {
-    public static byte[] ToBigEndianByteArrayWithoutLeadingZeros(this long value)
+    public static ReadOnlySpan<byte> MutateToBigEndianSpanWithoutLeadingZeros(ref this long value)
```

Comment:
WHy does it mutate the original value? Is that safe?

Reply:
Its needs to return the Span over valid stackspace so it takes the long by ref and creates the span over it; so its valid

The wrinkle is it also needs to change its byte order from little endian to big endian; trashing the local variable on the other side. 

However there are only two instances where that's important; one I moved this call down one so is last use; and other I take a copy to a new local so the rest of the method can keep using the original value

Reply:
wouldn't it be better to always stackalloc 8 bytes?

Reply:
Can't return the stackalloc'd to the caller as it will have been unwound so no longer valid. Caller needs to allocate the space

Reply:
I could through in an out var and then mutate that instead so
```csharp
ReadOnlySpan<byte> ToBigEndianSpanWithoutLeadingZeros(this long value, out long bigEndian)
```

Then in use it would be:
```
key.MutateToBigEndianSpanWithoutLeadingZeros(out _)
```
but that's pretty horrible too

Reply:
I like this one a lot better - not to cut yourself. Looks good enough to me.

Reply:
Isn't a T constraint for `ReverseEndianness`; which does seem a missed opportunity for the runtime